lib/adyen/form.rb

```diff
@@ -234,16 +249,13 @@ def hidden_fields(parameters = {})
     # @param [Hash] parameters The parameters that will be included in the payment request.
     # @return [String] The string for which the siganture is calculated.
     def calculate_signature_string(parameters)
-      merchant_sig_string = ""
-      merchant_sig_string << parameters[:payment_amount].to_s       << parameters[:currency_code].to_s         <<
-                             parameters[:ship_before_date].to_s     << parameters[:merchant_reference].to_s    <<
-                             parameters[:skin_code].to_s            << parameters[:merchant_account].to_s      <<
-                             parameters[:session_validity].to_s     << parameters[:shopper_email].to_s         <<
-                             parameters[:shopper_reference].to_s    << parameters[:recurring_contract].to_s    <<
-                             parameters[:allowed_methods].to_s      << parameters[:blocked_methods].to_s       <<
-                             parameters[:shopper_statement].to_s    << parameters[:merchant_return_data].to_s  <<
-                             parameters[:billing_address_type].to_s << parameters[:delivery_address_type].to_s <<
-                             parameters[:shopper_type].to_s         << parameters[:offset].to_s
+      signature_params = PAYMENT_REQUEST_SIGNATURE_PARAMS.reduce({}) do |hash, param|
+        value = parameters[param]
+        hash[Adyen::Util.camelize(param.to_s)] = Adyen::Signature::escape_value(value.to_s) if parameters.key?(param)
```

Comment:
Also, I find it kind of confusing that you extract `value` out of `parameters`, but check in the next statement whether the parameter is set. It would make more sense to me to check this first.